src/kaggle/api/kaggle_api_extended.py

```diff
@@ -1740,41 +1751,67 @@ def dataset_create_version(self,
-        category_ids=keywords,
-        delete_old_versions=delete_old_versions)
+    body = ApiCreateDatasetVersionRequestBody()
+    body.version_notes=version_notes
+    body.subtitle=subtitle
+    body.description=description
+    body.files=[]
+    # body.convert_to_csv=convert_to_csv
+    body.category_ids=keywords
+    body.delete_old_versions=delete_old_versions
+
+    with self.build_kaggle_client() as kaggle:
+      if id_no:
+        request = ApiCreateDatasetVersionByIdRequest()
+        request.id = id_no
+        message = kaggle.datasets.dataset_api_client.create_dataset_version_by_id
+      else:
+        self.validate_dataset_string(ref)
+        ref_list = ref.split('/')
+        owner_slug = ref_list[0]
+        dataset_slug = ref_list[1]
+        request = ApiCreateDatasetVersionRequest()
+        request.owner_slug = owner_slug
+        request.dataset_slug = dataset_slug
+        message = kaggle.datasets.dataset_api_client.create_dataset_version
+      request.body = body
+      with ResumableUploadContext() as upload_context:
+        self.upload_files(body, resources, folder, ApiBlobType.DATASET,
+                          upload_context, quiet, dir_mode)
+        request.body.files = [self._api_dataset_new_file(file) for file in request.body.files]
+        response = self.with_retry(message)(request)
+        return response
 
     with ResumableUploadContext() as upload_context:
       self.upload_files(request, resources, folder, ApiBlobType.DATASET,
                         upload_context, quiet, dir_mode)
 
       if id_no:
         result = DatasetNewVersionResponse(
-            self.process_response(
-                self.with_retry(
-                    self.datasets_create_version_by_id_with_http_info)(
-                        id_no, request)))
+          self.process_response(
+            self.with_retry(
+              self.datasets_create_version_by_id_with_http_info)(
+              id_no, request)))
       else:
-        if ref == self.config_values[
-            self.CONFIG_NAME_USER] + '/INSERT_SLUG_HERE':
-          raise ValueError('Default slug detected, please change values before '
-                           'uploading')
+        if ref == self.config_values[self.CONFIG_NAME_USER] + '/INSERT_SLUG_HERE':
+          raise ValueError('Default slug detected, please change values before uploading')
         self.validate_dataset_string(ref)
         ref_list = ref.split('/')
         owner_slug = ref_list[0]
         dataset_slug = ref_list[1]
         result = DatasetNewVersionResponse(
-            self.process_response(
-                self.datasets_create_version_with_http_info(
-                    owner_slug, dataset_slug, request)))
+          self.process_response(
+            self.datasets_create_version_with_http_info(
+              owner_slug, dataset_slug, request)))
 
       return result
 
+  def _api_dataset_new_file(self, file):
+    # TODO Eliminate the need for this conversion

```

Comment:
is there a bug listing the blocker for this

Reply:
Not as such. The blocker is removing all Swagger code. That is, it is a note-to-self that I need to finish up a few things before calling it done.